lib/report_generators/search_term_by_repo_and_searched_from.rb

```diff
@@ -162,14 +173,86 @@ def generate_report!
 
   def upload_to_drive
     drive = Google::Apis::DriveV3::DriveService.new
-    scopes = ['https://www.googleapis.com/auth/drive.file']
-    auth = Google::Auth::ServiceAccountCredentials.make_creds(json_key_io: File.open(@auth_file, 'r'), scope: scopes)
+    drive.authorization = auth_drive
 
-    drive.authorization = auth
     # Upload a file
     metadata = Google::Apis::DriveV3::File.new(name: self.report_basename, mime_type: 'application/vnd.google-apps.spreadsheet')
     file = drive.create_file(metadata, upload_source: self.report_output_path, content_type: 'text/csv', supports_team_drives: true)
     drive.update_file(file.id, add_parents: @google_parent_id)
+
+    filter_spreadsheet(file)
+  end
+
+  def filter_spreadsheet(file)
+    sheets = Google::Apis::SheetsV4::SheetsService.new
+    sheets.authorization = auth_drive
+    
+    spreadsheet = sheets.get_spreadsheet(file.id)
+
+    requests = {
```

Comment:
This `requests` hash is ugly and verbose at the moment, but I'm not sure it's worth refactoring right now, as we may want to consider what a more manageable structure would look like (probably would rather specify the values to keep, not those to hide; remove the boiler plate `range`; etc., etc.).